### PR TITLE
Configure Expiration Date on CI images

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

The current images weren't being configured to expire. We have no need for the images created during our E2E testing after the tests have ended.

### CODE REVIEW

**Comment by chatton** [comment]
looks like this is not currently setting tag - investigating

File: scripts/ci/build_and_push_image.sh [hunk 1]
```diff
@@ -6,5 +6,7 @@
 echo "${quay_password:?}" | docker login "-u=${quay_user_name:?}" quay.io --password-stdin
 
 python3 scripts/dev/dockerfile_generator.py "${image_type:?}" > Dockerfile
-docker build . -f Dockerfile -t "${image:?}"
+
+# Providing the quay.expires-after configures quay to delete this image after the provided amount of time
+docker build . --label "quay.expires-after=${expire_after:-never}" -f Dockerfile -t "${image:?}"
```
Comment (rodrigovalin) [inline scripts/ci/build_and_push_image.sh:11]: 🥇

**Comment by rodrigovalin** [comment]
I approved without seeing the comment that the actual thing is not working :/

**Comment by chatton** [comment]
> I approved without seeing the comment that the actual thing is not working :/

It's working now 👍

**Pull request merged**